docs/content/docs/testing/devtools.md

```diff
@@ -0,0 +1,71 @@
+
+-   Container information and data visualization
+-   Ability to modify container state for testing offline and reconnect scenarios
+-   Visibility into audience membership, permissions, and join/leave logs
+-   Framework event logs to view performance and error events
+
+To use the Fluid DevTools, developers need to follow two simple steps:
+
+1. Integrate the Fluid DevTools library into your app using these steps: https://aka.ms/fluid/devtool
+    -   See DevTools integrated into samples at https://aka.ms/fluid/samples
+2. Install the Fluid DevTools browser extension from [Edge](https://aka.ms/fluid/devtool/edge) or [Chrome](https://aka.ms/fluid/devtool/chrome) extension stores
+
+Once you have completed the above steps, you can simply launch your Fluid application in a browser window and start the browser DevTools by right-clicking and selecting `Inspect`. In the browser DevTools window, you should see a tab for `Fluid Developer Tools`
+
+## Container Information
+
+DevTool shows all the containers in your application, their current state (connected, attached, disconnected) and allows developers to modify the state (disconnect, close). Developers can also see a lot of container state changes with timestamps.
+
+![A screenshot showing container information](/images/container_info.png)
+
+-   **Disconnect/Reconnect Container** button disconnects it from the Fluid service
+-   **Close Container** button detaches from the Fluid service
+
+![A screenshot showing container state changes](/images/container_states.png)
+
+These buttons can be used to test the following scenarios:
+
+-   **Offline state** - Test how your app will behave if connection drops in the middle of a collab session
+-   **Merge edge cases** - Test how your app reacts to various merging scenarios when multiple users work offline and come back online
+-   **Container cleanup** - Test how your app behaves when a container is closed and disposed off.
+
+## Data Visualization
+
+DevTool shows the current container data including all the DDSs that are part of the container. Currently, the DevTool supports renderers for `SharedTree`, `SharedMap`, `SharedString`, `SharedCounter` and `Sequences`. Because of the extensible model of the framework, you can also build and add renderers for custom DDSs.
+
+![A screenshot showing container data visualization](/images/container_data_viz.png)
+
+The DDS visualizer updates in real-time as local and remote changes impact the container data. Getting a peek at the underlying data and checking it against what's rendered in your app should allow you to look for any inconsistencies between the data and your app UI.
+
+## Audience Info
+
+DevTool displays a list of users that are connected to the container(s) in the application and their permission level (read/write). The tool also shows a log of users joining and leaving the container. This can be used to debug user presence indicators in your apps.
```

Comment:
For extra clarity. I don't recall that we try to reach into any user-provided information about the applications _users_, we just report on Fluid clients; and permissions could be misinterpreted, because a client whose user technically has write permissions to the container document in storage, could have a read connection right now.
```suggestion
DevTool displays a list of clients that are connected to the container(s) in the application and the kind of connection they have to it (read VS read/write). The tool also shows a log of clients joining and leaving the container. This can be used to debug user presence indicators in your apps.
```